common/metrics/defs.go

```diff
@@ -1399,6 +1399,14 @@ const (
 	NumWorkerScopes
 )
 
+// -- Operation scopes for ShardDistributor service --
+const (
+	// ShardDistributorGetShardOwnerScope tracks GetShardOwner API calls received by service
+	ShardDistributorGetShardOwnerScope = iota + NumCommonScopes
```

Comment:
nit: I wonder if we check there is no overlap? Metrics are defined globally, and having Scope following `NumCommonScopes` looks natural for other scopes as well.

Reply:
There is a test that all the consts here have a corresponding value in the map.

The numbers overlap for all history, matching, workers, and shard distributor by design, since one metrics client will serve one service.

I think it would be nice to refactor this for all of the repo. I'm not sure the Global definitions in general are helpful.